### DISCUSSION TITLE
Creating EventBus from Yaml or JSON programmatically

### DISCUSSION THREAD

**Discussion by srajeev** [argoproj/argo-events#2042] [General]
Hello -

How can we create EventBus  from yaml/json file programmatically (I am using Scala now)?
I mean I can create an Argo EventBus using kubectl apply -f test-eventbus.yaml
But how can I create the same programmatically from yaml or json file?

I was able to create sensor etc from a POST on  https://dev-daqe-argo-ui-aks.cds.test.net/api/v1/sensors/ with a JSON payload.

Thank you
Shla

**Comment by whynowy** [comment]
There's a java client, not sure if it's helpful for you. 

https://github.com/argoproj-labs/argo-events-client-java
